flutter/Graminsta/lib/auth/login.dart

```diff
@@ -1,11 +1,46 @@
 import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:Graminsta/http.dart';
+import 'package:dio/dio.dart';
 
 class LoginPage extends StatefulWidget {
   @override
   _LoginPageState createState() => new _LoginPageState();
 }
 
 class _LoginPageState extends State<LoginPage> {
+  final _usernameController = TextEditingController();
+  final _passwordController = TextEditingController();
+
+  void login() async {
+    Response response;
+    HttpManager.post("core/login/", data: {
+      "username": _usernameController.text,
+      "password": _passwordController.text,
+    }, success: (data) {
+      _setToken(data);
+    }, error: (e) {
+      Scaffold.of(context).showSnackBar(SnackBar(
+          content: Text('登录失败，请检查用户名和密码是否正确'),
+          duration: Duration(milliseconds: 2000)));
+      print(e);
+    });
+  }
+
+  _setToken(token) async {
+    final prefs = await SharedPreferences.getInstance();
+    final setTokenResult = await prefs.setString('user_token', token);
+    if (setTokenResult) {
+      //if set token success, redirect to homepage
+      Navigator.of(context).pushNamedAndRemoveUntil(
+        '/home',
+        (route) => route == null,
+      );
+    } else {
+      debugPrint('保存登录token失败');
+    }
+  }
```

Comment:
Put this function in a separate file，because it will also be used during registration.